app/javascript/components/generic-objects-form/index.jsx

```diff
@@ -0,0 +1,234 @@
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import get from 'lodash/get';
+import { TrashCan32 } from '@carbon/icons-react';
+import { Button } from 'carbon-components-react';
+
+const GenericObjectForm = ({ recordId }) => {
+  const [{ initialValues, isLoading }, setState] = useState({ isLoading: !!recordId });
+  const promise = useMemo(() => API.options('/api/generic_object_definitions/'), []);
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  //when the form is submitted, any uploaded image is encoded to Base64
+  const toBase64 = (file) =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = () => resolve(reader.result);
+      reader.onerror = (error) => reject(error);
+    });
+
+  //custom file edit component that displays the currently saved image and a delete button to replace it
+  const FileEditComponent = (props) => {
+    const { input, label, src, description } = useFieldApi(props);
+    const formOptions = useFormApi();
+
+    const deleteFile = () => {
+      formOptions.change('image_update', true);
+    };
+
+    return (
+      <div>
+        <label className='bx--label' htmlFor={input.name}>{label}</label>
+        <br />
+        <div className='edit-div'>
+          <img className='edit-image' id='imageDisplay' src={src} />
+          <Button className='edit-button' renderIcon={TrashCan32} iconDescription={description} hasIconOnly onClick={deleteFile} {...input} />
+        </div>
+      </div>
+    );
+  };
+
+  //custom validator that checks any uploaded files to see if they meet the requirements
+  const fileValidator = ({ maxSize }) => {
+    return (value) => {
+      const imageTypes = /image\/jpg|image\/jpeg|image\/png|image\/svg/;
+
+      if (value && value.inputFiles[0] && !imageTypes.test(value.inputFiles[0].type))
+        return __(`File must be an image of type "png", "jpg/jpeg", or "svg". The currently uploaded file's extension is "${value.inputFiles[0].type.split('/').pop()}"`);
+
+      if (value && value.inputFiles[0] && value.inputFiles[0].size > maxSize)
+        return __(`File is too large, maximum allowed size is ${maxSize} bytes. Current file has ${value.inputFiles[0].size} bytes`);
+    };
+  };
+
+  //custom validator that makes sure attribute, association, and method names can only contain lowercase letters, numbers or underscores
+  const syntaxValidator = () => {
+    var format = /[ `!@#$%^&*()+\-=\[\]{};':"\\|,.<>\/?~]/;
+    return (value) => {
+      if (value != value.toLowerCase() || format.test(value))
+        return __(`Name can only contain lowercase letters, numbers, or underscores`);
+    };
+  };
+
+  //custom validator that checks to make sure no two attributes/associations/methods have the same name
+  const uniqueNameValidator = (values) => {
+    const errors = { attributes: [], associations: [], methods: [] };
+
+    for (const index in values.attributes) {
```

Comment:
If i am not wrong, All these 3 functions similar except arguments changing, can you move logic into a function?